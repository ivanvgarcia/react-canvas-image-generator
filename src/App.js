import React, { useEffect } from 'react';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import styled, { createGlobalStyle } from 'styled-components';
import ScrollToTop from 'components/ScrollToTop';
import Landing from 'components/landing/Landing';
import Canvas from 'components/canvas/Canvas';
import Dashboard from 'components/dashboard/Dashboard';
import Navigation from 'components/navigation/Navigation';
import PrivateRoute from 'components/routing/PrivateRoute';
import { Provider } from 'react-redux';
import store from './store.js';
import { checkSession } from './actions/auth';

const GlobalStyle = createGlobalStyle`
  html, body {
    height: 100%;
  }
  body {
    margin: 0;
    padding: 0;
    border-sizing: border-box;
    font-family: "Helvetica";
    height: 100%;
  }
  #root{
    height: 100%;
  }
  .App {
    height: 100%;
  }
  .slick-slider .slick-prev {
    z-index: 999 ;
    left: 5px ;
  }
  .slick-slider .slick-next {
    z-index: 999 ;
    right: 5px ;
  }
  h1 {
    color: white;
  }
`;

const Container = styled.div`
  height: 100%;
`;

function App() {
  useEffect(() => {
    if (!store.getState().auth.user) {
      store.dispatch(checkSession());
    }
  }, []);

  return (
    <Router className="App">
      <Provider store={store}>
        <ScrollToTop>
          <GlobalStyle />
          <Container>
            <Navigation />
            <Switch>
              <Route exact path="/" component={Landing} />
              <Route exact path="/avatar-generator" component={Canvas} />
              <PrivateRoute exact path="/dashboard" component={Dashboard} />
            </Switch>
          </Container>
        </ScrollToTop>
      </Provider>
    </Router>
  );
}

export default App;
