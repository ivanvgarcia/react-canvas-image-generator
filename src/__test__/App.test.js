import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';
import 'setupTest.js';
import App from 'App';

afterEach(cleanup);

it('matches snapshot', () => {
  const { asFragment } = render(<App />);
  expect(asFragment()).toMatchSnapshot();
});

it('it displays the title', () => {
  const { getByTestId } = render(<App />);
  const heading = getByTestId('title');
  expect(heading).toBe('Cocoppa Generator');
});
