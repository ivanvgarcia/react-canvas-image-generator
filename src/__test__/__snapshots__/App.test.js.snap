// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-brqgnP joMbJc"
  >
    <ul
      class="sc-hzDkRC XkaRE"
    >
      <li
        class="sc-jhAzac csYfOn"
      >
        <svg>
          menu.svg
        </svg>
      </li>
    </ul>
    <div
      class="sc-bdVaJa Zgpxm"
    >
      <h1
        class="sc-gzVnrw jCFwwF"
      >
        landing.title
      </h1>
      <a
        href="/avatar-generator"
      >
        landing.cta-button
      </a>
      <p
        style="text-align: center; color: white; font-size: 1.2rem;"
      >
        - or -
      </p>
      <button
        class="sc-fjdhpX gbmzxk"
      >
        <img
          alt="twitter"
          src="images/twitter.png"
        />
        landing.login
      </button>
    </div>
  </div>
</DocumentFragment>
`;
